src/test/clojure/clara/performance/test_compilation.clj

```diff
@@ -0,0 +1,154 @@
+                          (time-execution func))
+        sum #(reduce + %)
+        mean (/ (sum execution-times) iterations)
+        std (->
+              (into []
+                    (comp
+                      (map #(- % mean))
+                      (map #(Math/pow (double %) 2.0)))
+                    execution-times)
+              sum
+              (/ iterations)
+              Math/sqrt)]
+    {:std (double std)
+     :mean (double mean)}))
+
+(defn run-performance-test
+  [form]
+  (let [{:keys [description func iterations mean-assertion]} form
+        {:keys [std mean]} (execute-tests func iterations)]
+    (println (str \newline "Running Performance tests for:"))
+    (println description)
+    (println "==========================================")
+    (println (str "Mean: " mean "ms"))
+    (println (str "Standard Deviation : " std "ms" \newline))
+    (is (mean-assertion mean))))
+
+(def base-production
+  {:ns-name (symbol (str *ns*))})
+
+(defn generate-filter-productions
+  [seed-syms]
+  (into {}
+        (for [seed-sym seed-syms
+              :let [next-fact (symbol (str seed-sym "prime"))
+                    production (assoc base-production
+                                 :lhs [{:type (keyword seed-sym)
+                                        :constraints ['(= this ?binding) `(filter-fn ~'this)]}]
+                                 :rhs `(r/insert! (with-meta ~(set (repeatedly 10 #(rand-nth (range 100))))
+                                                             {:type ~(keyword next-fact)
+                                                              :val ~'?binding})))]]
+          [next-fact production])))
+
+(defn generate-compose-productions
+  [seed-syms]
+  (let [template (fn template
+                   ([l] (template l l))
+                   ([l r] (let [next-fact (symbol (str l r "prime"))
+                                production (assoc base-production
+                                             :lhs [{:type (keyword l)
+                                                    :constraints ['(= this ?binding-l)]}
+                                                   {:type (keyword r)
+                                                    :constraints ['(= ?binding-l this)]}]
+                                             :rhs `(r/insert! (with-meta ~(set (repeatedly 10 #(rand-nth (range 100))))
+                                                                         {:type ~(keyword next-fact)})))]
+                            [next-fact production])))]
+    (into {}
+          (for [combo (partition-all 2 (shuffle seed-syms))]
+            (apply template combo)))))
+
+(defn generate-collection-productions
+  [seed-syms]
+  (into {}
+        (for [seed-sym seed-syms
+              :let [next-fact (symbol (str seed-sym "prime"))
+                    production (assoc base-production
+                                 :lhs [{:accumulator '(clara.rules.accumulators/all)
```

Comment:
`clara.rules.accumulators` needs to be required explicitly by the ns running these rules, which I believe is this one. There shouldn't be assumptions about other namespaces already requiring your direct dependencies. That can lead to compilation-order dependent runs that fail occasionally.